route/build/src/mpi_utils.f90

```diff
@@ -334,6 +335,58 @@ SUBROUTINE shr_mpi_bcastLogical(allocArray,    & ! inout: array to be broadcaste
 
   END SUBROUTINE shr_mpi_bcastLogical
 
+  ! ----------------------------------
+  ! BROADCAST - character allocatable array (fixed character length)
+  ! ----------------------------------
+  SUBROUTINE shr_mpi_bcastChar(allocArray,    & ! inout: array to be broadcasted to each proc
+                               ierr, message)   ! output: error handling
+    implicit none
+    ! Argument variables:
+    character(strLen), allocatable, intent(inout) :: allocArray(:) ! inout: array to be sent to proc
+    integer(i4b),                   intent(out)   :: ierr
+    character(strLen),              intent(out)   :: message       ! error message
+    ! local variable
+    character(len=:), allocatable                 :: flat_str
+    integer(i4b)                                  :: flat_size
+    integer(i4b)                                  :: idx
+    integer(i4b)                                  :: num_elements
+
+    ierr=0; message='shr_mpi_bcastChar/'
+
+    if (masterproc) then
```

Comment:
Suggest following how this is done in csm_share.

https://github.com/ESCOMP/CESM_share/blob/main/src/shr_mpi_mod.F90